File: app/src/main/java/com/example/myapplication/SelectTemplateFragment.java
package com.example.myapplication;

import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class SelectTemplateFragment extends Fragment {

    RecyclerView recyclerView;
    private List<Select_Template_Item> selectTemplateItems;
    private SelectTemplateAdapter selectTemplateAdapter;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.selecttemplatefragment, null);

        Toolbar toolbar = getActivity().findViewById(R.id.toolbar);
        toolbar.setTitle("Create New Story");

        recyclerView = view.findViewById(R.id.select_template_recyclerview);
        recyclerView.setHasFixedSize(true);
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));

        selectTemplateItems = new ArrayList<>();

        Select_Template_Item listItems = new Select_Template_Item("1", BitmapFactory.decodeResource(getContext().getResources(),
                R.drawable.ic_launcher_background));
        selectTemplateItems.add(listItems);

        Select_Template_Item listItems1 = new Select_Template_Item( "2",BitmapFactory.decodeResource(getContext().getResources(),
                R.drawable.ic_launcher_background));
        selectTemplateItems.add(listItems1);

        Select_Template_Item listItems2 = new Select_Template_Item("3",BitmapFactory.decodeResource(getContext().getResources(),
                R.drawable.ic_launcher_background));
        selectTemplateItems.add(listItems2);

        //Collections.reverse(myStoryListItems);

        selectTemplateAdapter = new SelectTemplateAdapter(getContext(), selectTemplateItems);

        recyclerView.setAdapter(selectTemplateAdapter);

        return view;
    }

}

File: app/src/main/java/com/havrtz/openworld/fragments/SelectTemplateFragment.kt
package com.havrtz.openworld.fragments

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.ViewTreeObserver
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.havrtz.openworld.adapters.SelectTemplateAdapter
import com.havrtz.openworld.models.Template
import com.havrtz.openworld.R
import com.havrtz.openworld.helpers.ColumnSizeCalculator
import com.havrtz.openworld.helpers.EqualSpacingItemDecoration
import com.havrtz.openworld.viewmodels.TemplateViewModel

class SelectTemplateFragment : Fragment() {
    lateinit var recyclerView: RecyclerView
    private lateinit var templateViewModel: TemplateViewModel

    @SuppressLint("InflateParams")
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.select_template_fragment, null)

        recyclerView = view.findViewById(R.id.select_template_recyclerview)

        recyclerView.viewTreeObserver.addOnGlobalLayoutListener(object : ViewTreeObserver.OnGlobalLayoutListener {
            override fun onGlobalLayout() {
                view.viewTreeObserver.removeOnGlobalLayoutListener(this)

                // get width and height of the view
                recyclerView.layoutManager = GridLayoutManager(context, 2)
                (recyclerView.layoutManager as GridLayoutManager).spanCount = ColumnSizeCalculator.calculateSize(recyclerView.width)
            }
        })

        recyclerView.addItemDecoration(EqualSpacingItemDecoration(16, EqualSpacingItemDecoration.HORIZONTAL))
        recyclerView.setHasFixedSize(true)

        val selectTemplateAdapter = context?.let { SelectTemplateAdapter(it) }
        recyclerView.adapter = selectTemplateAdapter
        
        /** Get story view model and show data  */
        templateViewModel = ViewModelProvider(this).get(TemplateViewModel::class.java)
        templateViewModel.alltemplates?.observe(viewLifecycleOwner, Observer<List<Template?>?> { templates ->
            selectTemplateAdapter?.setTemplates(templates as List<Template>)
        })
        /** Get story view model and show data  */

        return view
    }
}